### ISSUE TITLE
run_ipmnist / run_label_emnist accept out-of-range, non-integer labels and non-finite inputs; JAX clamps the gather and the run reports a plausible in-range accuracy

### ISSUE THREAD

**Issue by Prabhat1308** [SlopDotCash/asi#447]
## Summary

Neither full-horizon protocol runner validates its label domain or input finiteness. `run_ipmnist` (`alberta_framework/benchmarks/upgd_ipmnist.py`) and `run_label_emnist` (`alberta_framework/benchmarks/upgd_label_emnist.py`, whose docstring states "`data_y`: `(n_train,)` integer class labels in `[0, n_classes)`") check `data_x.shape[1] == input_dim` and `data_y.shape == (n,)` and stop; `data_y` is coerced with `jnp.asarray(..., int32)` (truncating floats) and used as a gather index into the length-`n_classes` softmax. JAX out-of-bounds gathers **clamp**, so every label `>= n_classes` is silently scored and trained as class `n_classes-1`; non-finite `data_x` is never rejected, and `argmax` of an all-NaN logit row returns 0, so a fully poisoned forward pass still yields a finite in-range accuracy. The screening lane already has the guard (`ipmnist_screening._validated_ipmnist_data`: integer dtype, non-negative, `< n_classes`, finite `data_x`) — the runners it wraps do not.

Reachability: on the label lane `load_plan` never pins `n_classes == 47`, so a plan with `n_classes = 10` folds 37 real EMNIST classes onto class 9 and reports an inflated "online accuracy" against the paper's read-off; the bad run raises nothing, and `partial_payload` serializes the in-range curve.

## Falsifier (deterministic, pristine `main` @ `b8a37de`)

```
jnp clamp check: jnp.arange(3)[5] = 2
IPMNIST run with labels==7 and n_classes==3: NO ERROR
  per_task_accuracy(out-of-range) = [[0.25, 0.0, 0.25]]  == identical to clipping labels to 2: True
  partial_payload accepts it: alberta.upgd_ipmnist.partial.v2
LabelEMNIST run with labels==7 and n_classes==3: NO ERROR  (identical to clamped run: True)
float label 2.9 truncated to 2 without error: True
run_ipmnist with all-inf data_x: NO ERROR, per_task_accuracy = [[1.0, 1.0]]  loss = [[nan, nan]]
ipmnist_screening._validated_ipmnist_data (same inputs): "data_y class labels must be smaller than 3" / "data_x must contain only finite values"
```

Expected: both runners reject non-integer labels, labels outside `[0, n_classes)`, and non-finite inputs before learner setup, using the one shared validator.

## Dedup

Same class, other surfaces: #398/#399 (PrototypeAgent silent reshape), #403/#404 (init_cbp_state widths), #340/#341 and #442/#444 (curve domains). `gh issue/pr list --search "run_ipmnist labels" / "n_classes" / "EMNIST" / "load_mnist"`: nothing on the runners' label domain. Related open (mine, same file, different region): #442/#444; merge-tree composes cleanly.

Fix PR follows (failing-test-first).
